proposals/lip-0061.md

```diff
@@ -418,8 +418,8 @@ If a validator node starts for the first time or loses all previously created si
 
 If for a validator `v`, after applying a block the value returned by the function `getBFTHeights().maxHeightPrecommitted` increases from `h1` to `h2`, then the following commit messages are created by the validator `v` and gossiped as described further below:
```

Comment:
Here I would define `h1 = max(getMaxRemovalHeight(), previous maxHeightPrecommitted)` for the following reasons:

1) Efficiency: Since we don't want to allow commits for height smaller than `MIN_CERTIFICATE_HEIGHT`, it would be really waste of resources to loop through all blocks of init rounds (roughly 60k blocks) with no reason. 
2) Clarity: It will make the whole part (together with initial single commit creation) much cleaner and easier to parse.

Reply:
Note that you can equivalently do `h1 = max(MIN_CERTIFICATE_HEIGHT - 1, previous maxHeightPrecommitted)`. This avoid calling `getMaxRemovalHeight()` and additional state reads. I updated the specs in this direction.

Reply:
Fine. Just to make sure the reasoning, is this because `previous maxHeightPrecommitted >= b.header.aggregateCommit.height` should always hold, right?

Reply:
In general, for every block we have `maxHeightPrecommited (after processing block) >= b.header.aggregateCommit.height`. Further, we have that the finalized height is strictly lower than the finalized height. Hence,
```
maxHeightPrecommitted at currentHeight -1 >= b.header.aggregateCommit.height at currentHeight -1 >= b.header.aggregateCommit.height at maxHeightFinalized
```
Therefore,
```
 max(MIN_CERTIFICATE_HEIGHT - 1, maxHeightPrecommitted at currentHeight -1)  >= getMaxRemovalHeight()
```

Reply:
Ok, this is exactly what I meant by `previous maxHeightPrecommitted >= b.header.aggregateCommit.height` (meaning b at maxHeightFinalized as in the `getMaxRemovalHeight` function).

Thanks for the clarification!